### ISSUE TITLE
Power-On Errors Out

### ISSUE THREAD

**Issue by eddcgarcia** [hashicorp/terraform-provider-vsphere#945]
When running `terraform apply`, I am getting this error:
```
error powering on virtual machine: Post https://vsphere-server-xyz/sdk: context deadline exceeded
```

The reason this is occurring is because on power-up, there are vsphere questions coming up (https://docs.vmware.com/en/VMware-vSphere/6.7/com.vmware.vsphere.vm_admin.doc/GUID-50FCD2F8-FC99-4AE1-B839-1AC47BF884DC.html), and so the vm does not turn on before the `defaultAPITimeout` hits, which is currently 5 minutes. It does eventually turn on, but by then, terraform has errored out.

Is it possible to have a global `api_timeout` as was suggested here: https://github.com/terraform-providers/terraform-provider-vsphere/pull/162. Or even just being able to specify a longer wait for boot up specifically would be nice.

### Terraform Version
Terraform v0.12.9

### Affected Resource(s)
- `vsphere_virtual_machine`

### Important Factoids
Virtual machine questions come up on boot (https://docs.vmware.com/en/VMware-vSphere/6.7/com.vmware.vsphere.vm_admin.doc/GUID-50FCD2F8-FC99-4AE1-B839-1AC47BF884DC.html).

### References
- GH-162

**Comment by koikonom** [comment]
Hi @eddcgarcia, can you please share the questions you get while booting up a VM?

**Comment by eddcgarcia** [comment]
> Cannot connect the virtual device serial0 because no corresponding device is available on the host. Do you want to try to connect this virtual device every time you power on the virtual machine?

> Cannot connect the virtual device sound because no corresponding device is available on the host. Do you want to try to connect this virtual device every time you power on the virtual machine?

They're both yes or no questions.

**Comment by koikonom** [comment]
Thanks @eddcgarcia, we already have some logic to handle questions, it's just a matter of adding these and deciding what's the best way to handle it from a configuration perspective. 

For example, something along the lines of `always_try_connecting = ["sound", "serial0"]` .

**Comment by tenthirtyam** [comment]
Resolved in #1405 with the introduction of the `defaultAPITimeout` configuration for the provider.

Marking this issue as closed. If this issue is reappears with the latest version of the provider, please create a new issue linking back to this one for added context.

**Ryan Johnson**
Staff II Solutions Architect
Cloud Infrastructure Business Group, VMware
